x-pack/plugins/log_explorer/public/components/dataset_selector/dataset_selector.tsx

```diff
@@ -85,92 +83,147 @@ export function DatasetSelector({
-        />
-      ),
+      id: UNCATEGORIZED_TAB_ID,
+      name: uncategorizedLabel,
+      onClick: () => {
+        onStreamsEntryClick(); // Lazy-load uncategorized datasets only when accessing the Uncategorized tab
+        switchToUncategorizedTab();
+      },
+      'data-test-subj': 'datasetSelectorUncategorizedTab',
     },
-    ...integrationPanels,
   ];
 
+  const tabEntries = tabs.map((tab) => (
+    <EuiTab
+      key={tab.id}
+      onClick={tab.onClick}
+      isSelected={tab.id === tabId}
+      data-test-subj={tab['data-test-subj']}
+    >
+      {tab.name}
+    </EuiTab>
+  ));
+
   return (
     <DatasetsPopover
       selection={datasetSelection.selection}
       isOpen={isOpen}
       closePopover={closePopover}
       onClick={togglePopover}
     >
+      <Tabs>{tabEntries}</Tabs>
+      <SelectorActions>
+        <SelectorActions.ShowAllLogs isSelected={isAllMode} onClick={selectAllLogDataset} />
+      </SelectorActions>
+      <EuiHorizontalRule margin="none" />
       <SearchControls
         key={panelId}
         search={search}
         onSearch={searchByName}
         onSort={sortByOrder}
-        isLoading={isLoadingIntegrations || isLoadingStreams}
+        isLoading={isSearchingIntegrations || isLoadingStreams}
       />
       <EuiHorizontalRule margin="none" />
-      <EuiContextMenu
+      {/* For a smoother user experience, we keep each tab content mount and we only show the select one
+      "hiding" all the others. Unmounting mounting each tab content on change makes it feel glitchy,
+      while the tradeoff of keeping the contents in memory provide a better UX. */}
+      {/* Integrations tab content */}
+      <ContextMenu
+        hidden={tabId !== INTEGRATIONS_TAB_ID}
         initialPanelId={panelId}
-        panels={panels}
+        panels={[
+          {
+            id: INTEGRATIONS_PANEL_ID,
+            title: integrationsLabel,
+            width: DATA_VIEW_POPOVER_CONTENT_WIDTH,
+            items: integrationItems,
+          },
+          ...integrationPanels,
+        ]}
         onPanelChange={changePanel}
         className="eui-yScroll"
-        css={contextMenuStyles}
-        data-test-subj="datasetSelectorContextMenu"
+        data-test-subj="integrationsContextMenu"
+        size="s"
+      />
+      {/* Uncategorized tab content */}
+      <ContextMenu
+        hidden={tabId !== UNCATEGORIZED_TAB_ID}
+        initialPanelId={UNCATEGORIZED_PANEL_ID}
+        panels={[
+          {
+            id: UNCATEGORIZED_PANEL_ID,
+            title: uncategorizedLabel,
+            width: DATA_VIEW_POPOVER_CONTENT_WIDTH,
+            items: uncategorizedItems,
+          },
+        ]}
+        className="eui-yScroll"
+        data-test-subj="uncategorizedContextMenu"
         size="s"
       />
     </DatasetsPopover>
   );
 }
+
+const Tabs = styled(EuiTabs)`
+  padding: 0 8px;
+`;
+
+const ContextMenu = styled(EuiContextMenu)`
+  max-height: 440px;
+  transition: none !important;
```

Comment:
**note**: setting `!important` is not ideal, but to override the transition of the `EuiContextMenu` component it is necessary to go this way.

**Why?**
I'm removing this transition as a tradeoff. It gives a nice effect when the list is resized on its content change, but when the menu is fully hidden, its `height` is set to 0px, resulting in a glitchy effect when switching between tabs.
I opted for having a visually stable switch against the UI effect, but I'm open to rethink this if there different opinions.

**With transition (glitchy)**

https://github.com/elastic/kibana/assets/34506779/797b12bb-121e-49bd-b8fd-834644a41b1b

**Without transition**

https://github.com/elastic/kibana/assets/34506779/01c97b33-8921-4a47-aca1-fa19fe9213a8

Reply:
Thanks for the explanation here 👍